test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,344 @@
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+            configuration.Namespaces.Should().HaveCount(1);
+
+            // No match
+            var packageSourcesMatchPartial1 = configuration.GetConfiguredPackageSources("Cont");
+            Assert.Null(packageSourcesMatchPartial1);
+
+            // No match
+            var packageSourcesMatchPartial2 = configuration.GetConfiguredPackageSources("Contoso.Opensource");
+            Assert.Null(packageSourcesMatchPartial2);
+
+            // Match
+            var packageSourcesMatchFull1 = configuration.GetConfiguredPackageSources("Contoso.Opensource.MVC");
+            Assert.Equal(1, packageSourcesMatchFull1.Count);
+            Assert.Equal("publicrepository", packageSourcesMatchFull1.First());
+
+            // Match
+            var packageSourcesMatchFull2 = configuration.GetConfiguredPackageSources("Contoso.Opensource.MVC.ASP");
+
+            Assert.Equal(1, packageSourcesMatchFull2.Count);
+            Assert.Equal("publicrepository", packageSourcesMatchFull2.First());
+
+            // No match
+            var packageSourcesNoMatch = configuration.GetConfiguredPackageSources("random");
+            Assert.Null(packageSourcesNoMatch);
+        }
+
+        [Fact]
+        public void SearchTree_WithMultipleSources()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""nuget.org"">
+            <namespace id=""stuff"" />
+        </packageSource>
+        <packageSource key=""contoso"">
+            <namespace id=""moreStuff"" />
+        </packageSource>
+        <packageSource key=""privateRepository"">
+            <namespace id=""private*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+            configuration.Namespaces.Should().HaveCount(3);
+
+            var packageSourcesMatchFull1 = configuration.GetConfiguredPackageSources("stuff");
+
+            Assert.Equal(1, packageSourcesMatchFull1.Count);
+            Assert.Equal("nuget.org", packageSourcesMatchFull1.First());
+
+            var packageSourcesMatchPartial1 = configuration.GetConfiguredPackageSources("stu");
```

Comment:
Using a theory to provide data might allow these tests to be more readable and more extensive.

Reply:
I agree. But currently `PackageNamespacesConfiguration` only accept data from `ISettings` to create instance.
So I need create factory/test fixture for normal string input to make it work. Currently while api still taking shape creating it and maintaining different unit tests takes much more time. Maybe I'll do with when I implement `-source` changes.

Reply:
There's an internal constructor. 

Please use that instead: https://github.com/NuGet/NuGet.Client/blob/a696568069810de3912e712fda21ea7c987793f0/src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs#L17-L20